# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/base/TextLeafRange.h
# Commit: 16aee29be1b7
# Full Hash: 16aee29be1b7ae71eae277b97813fe6bf851ac0f
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-05-27 15:58:57
# Regressor Bug: 1737919
# File Overlap Count: 1
# Description:
#   Bug 1737919 part 1: TextLeafPoint::FindTextAttrsStart: Don't allow the caller to supply origin attributes. r=morgan
#   
#   This was done for LocalAccessibles to avoid the need to fetch attributes twice when we're simultaneously fetching boundaries and returning the attributes to a client.
#   However, GetTextAttributes will soon handle spelling errors, so it will no longer return the same data as GetTextAttributesLocalAcc (which doesn't handle spelling errors).
#   This breaks the equality checks in FindTextAttrsStart.
# ==============================================================================

diff -r 27f0da23c937 -r 16aee29be1b7 accessible/base/TextLeafRange.h
--- a/accessible/base/TextLeafRange.h	Fri May 27 04:32:40 2022 +0000
+++ b/accessible/base/TextLeafRange.h	Fri May 27 10:56:40 2022 +0000
@@ -135,18 +135,9 @@
    * (depending on the direction).
    * If aIncludeorigin is true and this is at a boundary, this will be
    * returned unchanged.
-   * aOriginAttrs allows the caller to supply the text attributes for this (as
-   * retrieved by GetTextAttributes). This can be used to avoid fetching the
-   * attributes twice if they are also to be used for something else; e.g.
-   * returning the attributes to a client. If aOriginAttrs is null, this method
-   * will fetch the attributes itself.
-   * aIncludeDefaults specifies whether aOriginAttrs includes default
-   * attributes.
    */
   TextLeafPoint FindTextAttrsStart(nsDirection aDirection,
-                                   bool aIncludeOrigin = false,
-                                   const AccAttributes* aOriginAttrs = nullptr,
-                                   bool aIncludeDefaults = true) const;
+                                   bool aIncludeOrigin = false) const;
 
   /**
    * Returns a rect (in dev pixels) describing position and size of